interface-spec.md

```diff
@@ -38,9 +43,7 @@ TODO: to what extent should we use typed signatures (`.equals(Node other)`) vers
 
 **Properties:**
 
-- `String .value` blank node name as a string (example: `_:blank3`)
-
-TODO: Does the value always start with an underscore?
+- `String .value` blank node name as a string without leading `_:` (example: `blank3`)
```

Comment:
> So doesn't that make the _: a property of the syntax?

The whole identifier is a property of the syntax.

> [combination of two different syntaxes example]
> That's a bit heretical but quite practical. It also illustrates that the _: is just Turtle's syntax for flagging a bnode.

Well, I think that example is bit presumptuous, but I do understand the practicality of it. I do think it's quite useful to preserve blank node identifiers (eg: for representing canonicalized data sets, for example) even if they aren't in the abstract data model. But there's a consistency issue based on the current wording.

You can make a similar example:

```
  RDF/XML: <rdf:Description rdf:nodeId="_:proteinProductTP53"><rdfs:label>P53 product</rdfs:label></rdf:Description>
+ Turtle:  _:proteinProductTP53 a :GeneProduct .
= _:proteinProductTP53 a :GeneProduct ; rdfs:label "P53 product" .
```

And I believe it's just as "correct" if we're consistently following the wording given. This would draw the opposite conclusion, that `_:` is not **just Turtle's** syntax for flagging a bnode. We're applying `_:` to all source syntaxes.

**Just imagine if we parse `<... rdf:nodeId="_:b">` and <... rdf:nodeId="b">` from the same document. Are we going to have a collision that mangles the dataset using the rules laid out above?**

In Turtle, blank nodes are described as the combination of `_:` and a "blank node label". And it is true that two blank nodes having the same label are in fact to be considered the same node. That does seem to make them *identifiers* within that syntax. However, that doesn't mean that this rule extends across other syntaxes. RDF/XML has no concept of "blank node labels", only "blank node identifiers".

So, the real problem here is that we're taking syntax-specific rules and leaking them into our representation of the data model. However, our representation can't preserve the graph relationships without blank node identifiers. This means we could either generate fresh ones -- or we can try and preserve the ones from the concrete syntax from which we loaded the dataset. Again, I do actually think it's important to do so because it makes canonicalizing data sets possible.

So, all of that is to say that I think perhaps my issue can be resolved by reconsidering the wording "blank node name as a string without leading `_:`". This presumes too much about the source concrete syntax. We need some other wording that is more general or perhaps we could address it via a parenthetical:

"blank node identifier as a string (note: for clarity, if the identifier was parsed from a concrete syntax that prefixes blank node identifiers with `_:`, this prefix is removed).

This would make it clear that when you parse `rdf:nodeId` from RDF/XML, if the value is taken *as is*, even if `_:` is a prefix. If you parse from Turtle, `_:` is removed. This should avoid mangling data.